### ISSUE TITLE
Can't upload my submarines into steam workshop

### ISSUE THREAD

**Issue by endkinghenry** [FakeFishGames/Barotrauma#12040]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

When i try to upload a file to steam workshop, it used to work in the past. But now for some reason, I'm unable to update or upload my own submarine creations. It shows "Publishing (file name) In the steam workshop failed. The 

![image](https://github.com/Regalis11/Barotrauma/assets/47675233/2446b210-7b47-4ef4-9618-efad0484a8ad)

preview is 552 kb,

Whats weird is, when i look into my steam workshop page, it tells me it is created but it's empty

![image](https://github.com/Regalis11/Barotrauma/assets/47675233/cfa93530-a737-48ed-9108-1f6ba380997b)

### Reproduction steps

1. Go to mods
2. Go to publish
3. Find your submarine you want to upload
4. Press apply, Have no preview, description and change note. With tags of "submarine"
5. After pressing apply staring at "submitting item to the steamworkshop"
6. See error, "Publishing (file name) In steam workshop failed.
7. Feel bad because it used to work

### Bug prevalence

Happens every time I play

### Version

v1.0.13.2

### Which operating system did you encounter this bug on?

Windows

**Comment by Zipliks** [comment]
Same problem here, cannot publish if `Preview Image` is set.
Without image publishes successfully.

**Comment by giists** [comment]
After some more testing it seems that these are the steps to get the error.
1. upload a new submarine with a picture and description(shouldn't matter where the image is from)
2. scroll to the bottom and delete the item
3. go to publish it again with the same picture and description
4. error

**Comment by Regalis11** [comment]
@giists I think that's probably a separate issue, as the original issue didn't seem to cause the same error message to be thrown. Addressed in https://github.com/Regalis11/Barotrauma-development/commit/4f1bb5dd2467a6070e8cafad83ee005f2e1f2dbb

**Comment by giists** [comment]
Following the same steps as before I now get this error.
![image](https://github.com/Regalis11/Barotrauma/assets/35175875/c3a7c3ef-34f0-4cc7-98d1-92e573a7fbeb)

**Comment by PatrickJS44** [comment]
> I've found a solution on my issue, my router was blocking specific things which was partly disrupting the connection between steam and barotrauma. While it was creating a workshop content, it was empty due to the disruption. After solving the router blocking specific signals, everything seemed to work properly again.
> 
> Tldr, it was a weird network issue.




I am having exactly the same issue described by @enkinghenry @[endkinghenry].
Fail to publish my submarine in Steam workshop. The same Error pop up saying publishing the submarine in the Steam workshop failed.
My preview image was a .png, 114kb. The publishing failed with or without the preview image, regardless of the directory of preview file. 
No red error lines in console. No warning or bug reported.

Empty mod would also appear on my steam workshop page. It created one barotrauma empty mod each time I tried to publish and failed.
![emptymod](https://github.com/user-attachments/assets/a4c40813-1fab-4031-a50c-fd810c2b7aec)

This issue made me recall that I had similar mod publishing problem two years ago when I played EA version of Barotrauma. I managed to publish my submarine with steamcmd then. After published successfully with steamcmd I could upgrade the mod normally with mod publishing menu built in game.
But these days I tried steamcmd and still failed (cannot even login with constant invalid password or no connection error).

I tried everything I could thought of. Reinstalled Steam, checked internet connection and firewall settings, the issue still there.
It would be grateful if anyone can give me a clue about what is going wrong or how to fix it.
